zetta_utils/bcube/bcube_strider.py

```diff
@@ -25,57 +27,209 @@ class BcubeStrider:
     Allows random indexing of the chunks without keeping the full chunk set in memory.
 
     :param bcube: Input bounding cube.
-    :param resolution: Resoluiton at which ``chunk_size`` and ``stride`` are given.
+    :param resolution: Resolution at which ``chunk_size`` and ``stride`` are given.
     :param chunk_size: Size of an individual chunk.
+    :param max_superchunk_size: Upper bound for the superchunks to consolidate the
+        individual chunks to. Defaults to ``chunk_size``.
     :param stride: Distance between neighboring chunks along each dimension.
+    :param stride_start_offset: Where the stride should start in unit (not voxel),
+        modulo ``chunk_size``.
+    :param mode: The modes that can be chosen for the behaviour.
+        `shrink` is the default behaviour, which will round the bcube down to
+        be aligned with the ``stride_start_offset`` (or just the start of the bcube
+        if ``stride_start_offset`` is not set or ``stride`` != ``chunk_size``).
+        `expand` is similar to `shrink` except it expands the bcube.
+        `exact` will give full cubes aligned with the ``stride_start_offset`` and the
+        ``chunk_size``, as well as the partial cubes at the edges.
+    :param silent: Suppresses all warnings.
```

Comment:
Is it better to let the default verbosity level handle this stuff? Maybe we can emit warnings as `info` or `debug`?

Reply:
I'm not sure - part of me thinks that any time the index changes inside a function we should emit an explicit warning, but using info or debug might be better. Your call.

Reply:
I think it's better to not have ad-hoc ways of handling verbosity

Reply:
Yeah, I guess if it has weird behaviour the user can just run `-v`. I'll change them to info.